src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/MQNotifierConfig.java

```diff
@@ -291,9 +312,26 @@ public void setVirtualHost(String virtualHost) {
     }
 
     /**
-     * Gets the routing key.
+     * Sets the routing key.
+     *
+     * @param routingKeyProvider the routing key.
+     */
+    public void setRoutingKeyProvider(String routingKeyProvider) {
+        this.routingKeyProvider = routingKeyProvider;
+    }
+    /**
+     * Gets the routing key provider.
+     *
+     * @return the routing key provider.
+     */
+
+    public String getRoutingKeyProvider() {
+        return this.routingKeyProvider;
+    }
+    /**
+     * Gets the routing key provider, either "auto" or "manual".
```

Comment:
```suggestion
     * Gets the routing key provider, either "AUTO" or "MANUAL".
```
Should we use constants for these values?